### PR TITLE
Rewrite with promises to accommodate for import shortstop resolution on every file load.

### PR DESCRIPTION
**Pull request by pvenkatakrishnan**

### CODE REVIEW

File: test/confit.js [hunk 1]
```diff
@@ -429,21 +425,56 @@ test('confit', function (t) {
     });
 
 
-    t.test('addOverride error', function (t) {
-        var basedir;
+    t.test('addOverride error with non existing filw', function (t) {
```
Comment (jasisk) [inline test/confit.js]: `non existing file`

Comment (pvenkatakrishnan) [inline test/confit.js]: fixed.

**Comment by pvenkatakrishnan** [comment]
will close this as we have a grand implementation on es6 of the same set of features

**Pull request closed without merging**